### ISSUE TITLE
[Masthead v2.1.0]: Phase 2 - Tiered menu tab order is horizontal, should be top to bottom

### ISSUE THREAD

**Issue by proeung** [carbon-design-system/carbon-for-ibm-dotcom#9337]
**Engineering info:**  
- JIRA Ticket: https://jsw.ibm.com/browse/ADCMS-2583
- JIRA Epic: https://jsw.ibm.com/browse/AEMSITES-177

-----

### Description
- A follow-up phase 2 task from https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/9310
- Tiered menu tab order is horizontal, should be top to bottom 
[See functional specs](https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=2325%3A136628)

![tab order Aug-30-2022 12-32-03](https://user-images.githubusercontent.com/45692622/187491269-ce022185-073f-4697-a3ee-c507ce9dec3a.gif)
<img width="1024" alt="Screen Shot 2022-08-30 at 12 34 07 PM" src="https://user-images.githubusercontent.com/45692622/187491529-71d35e02-d7a3-45bd-a4d8-cb93bce674d6.png">

### Design Specs
- Figma mockup - https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=2325%3A136628

**Comment by proeung** [comment]
@jkaeser Looks like you have the capability to work on this feature in this sprint and this is probably something that we can roll into phase 1 if feasible, however, let me know if that's not the case. Happy to keep this slated for Phase 2, if more work is involved.

**Comment by proeung** [comment]
Closing out this ticket as the work has already been covered in this PR (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9829)
